tests/CMakeLists.txt

```diff
@@ -161,6 +161,22 @@ foreach(testprog ${testprogs})
 endforeach()
 add_custom_target(testprogs DEPENDS ${compiled_testprogs})
 
+# Similarly compile all test libs
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/testlibs/)
+file(GLOB testlibs testlibs/*.c)
```

Comment:
I was going to suggest a convention like "starts with lib or lib_" but a separate dir actually seems cleaner :+1: